src/utils/helpers.js

```diff
@@ -77,6 +77,83 @@ var helpers = {
         return [hrz_altitude, hrz_azimuth]
     },
 
+    siderealTimetimesens: function(lon, date) {
+        /* Local Sidereal Time with reference to J2000
+        *(edited to accept date)
+        *
+        * Equations courtesy of www.stargazing.net/kepler/altaz.html 
+        * and www.aberdeenastro.org.uk/sidereal_time.htm
+
+        *  LST = 100.46 + 0.985647 * d + lon + 15*UT
+        *
+        *       d    is the days from J2000, including the fraction of a day
+        *       UT   is the universal time in decimal hours
+        *       lon is your longitude in decimal degrees, East positive.
+        */
+
+        var lmst, lon;
+        var epoch_date;
+        var today_time;
+        var epoch_time;
+        var milli_since_epoch;
+        var d, h, m, s;
+        var UT;
+
+        lon = parseFloat(lon)
+
+
+        // Calculate days since J2000
+        epoch_date = new Date(2000, 0, 1, 12, 0, 0);
+        
+        today_time = date.getTime();
+        epoch_time = epoch_date.getTime();
+
+        milli_since_epoch = today_time-epoch_time;
+        d = milli_since_epoch/86400000;
+
+        // Calculate UT: universal time in decimal hours
+        h = date.getUTCHours();
+        m = date.getUTCMinutes();
+        s = date.getUTCSeconds();
+        UT = h + m/60 + s/3600;
+
+        // Local Sidereal Time:
+        lmst = (100.46 + 0.985647*d + lon + 15*UT);
+        if (lmst<0) {
+            lmst = lmst +360
+        }
+
+        return lmst;
+    },
+    eq2altaztimesens: function (ra, dec, lat, lon, date ) {
```

Comment:
Full disclosure: my function/variable names are not the best either (and please let me know if you find any I should improve!)